### PR TITLE
Replaced _.has(object, property) with object.property

### PR DESCRIPTION
**Pull request by andreizet**
Addresses: #7747

Replaced all occurrences of ```_.has(object, property)``` with ```object.property```.
While trying to keep behaviour the same, in some cases ```object.property``` cannot be used. 

See the changes in:
• lib/plugins/aws/provider/awsProvider.js 
• lib/plugins/aws/package/compile/events/apiGateway/lib/validate.js

```javascript
    // This is false, caused by if(null)
    if(obj.prop)

    // This is true, since it only checks if 'prop' exists
    if(_.has(obj, 'prop'))

    // This is true, since it only checks if 'prop' exists
    if('prop' in obj)
```

I don't know if those two cases are isolated or not, but maybe another idea would be to replace ```object.property``` with ```'property' in object```.

### CODE REVIEW

File: lib/plugins/aws/deploy/lib/extendedValidate.js [hunk 1]
```diff
@@ -61,8 +61,7 @@ module.exports = {
       this.serverless.service.getAllFunctions().forEach(functionName => {
         const functionObject = this.serverless.service.getFunction(functionName);
         const individually =
-          (_.has(functionObject, ['package', 'individually']) &&
-            functionObject.package.individually) ||
+          (functionObject.package && functionObject.package.individually) ||
```
Comment (medikoo) [inline lib/plugins/aws/deploy/lib/extendedValidate.js:64]: Let's replace it with `_.get(functionObject, ['package', 'individually'])`

(Note that in future we will replace `_.get` usage with native optional chaining)

File: lib/plugins/aws/info/display.js [hunk 2]
```diff
@@ -15,7 +15,7 @@ module.exports = {
     message += `${chalk.yellow('stack:')} ${info.stack}\n`;
     message += `${chalk.yellow('resources:')} ${info.resourceCount}`;
 
-    if (_.has(info, 'resourceCount') && info.resourceCount >= 150) {
+    if (info.resourceCount && info.resourceCount >= 150) {
```
Comment (medikoo) [inline lib/plugins/aws/info/display.js:18]: This can be put just as `if (info.resourceCount >= 150)` (if there's no `info.resourceCount` it'll naturally be _false_)

File: lib/plugins/aws/deploy/lib/extendedValidate.js [hunk 3]
```diff
@@ -74,7 +73,7 @@ module.exports = {
           artifactFileName = this.provider.naming.getFunctionArtifactName(functionName);
           artifactFilePath = path.join(this.packagePath, artifactFileName);
 
-          if (_.has(functionObject, ['package', 'artifact'])) {
+          if (functionObject.package && functionObject.package.artifact) {
```
Comment (medikoo) [inline lib/plugins/aws/deploy/lib/extendedValidate.js:76]: Let's use `_.get`

File: lib/plugins/aws/deployFunction/index.js [hunk 4]
```diff
@@ -225,7 +225,7 @@ class AwsDeployFunction {
 
     // check if an artifact is used in function package level
     const functionObject = this.serverless.service.getFunction(this.options.function);
-    if (_.has(functionObject, ['package', 'artifact'])) {
+    if (functionObject.package && functionObject.package.artifact) {
```
Comment (medikoo) [inline lib/plugins/aws/deployFunction/index.js:228]: Let's use `_.get`

File: lib/plugins/aws/package/compile/events/apiGateway/lib/usagePlan.js [hunk 5]
```diff
@@ -32,8 +32,8 @@ function createUsagePlanResource(that, name) {
     } ${that.provider.getStage()} stage`;
     // assign quota
     if (
-      _.has(that.serverless.service.provider, 'usagePlan.quota') &&
-      that.serverless.service.provider.usagePlan.quota !== null
+      that.serverless.service.provider.usagePlan &&
+      that.serverless.service.provider.usagePlan.quota
```
Comment (medikoo) [inline lib/plugins/aws/package/compile/events/apiGateway/lib/usagePlan.js:36]: Let's use `_.get`

File: lib/plugins/aws/package/compile/events/apiGateway/lib/usagePlan.js [hunk 6]
```diff
@@ -47,8 +47,8 @@ function createUsagePlanResource(that, name) {
     }
     // assign throttle
     if (
-      _.has(that.serverless.service.provider, 'usagePlan.throttle') &&
-      that.serverless.service.provider.usagePlan.throttle !== null
+      that.serverless.service.provider.usagePlan &&
+      that.serverless.service.provider.usagePlan.throttle
```
Comment (medikoo) [inline lib/plugins/aws/package/compile/events/apiGateway/lib/usagePlan.js:51]: Let's use `_.get`

File: lib/plugins/aws/package/compile/events/apiGateway/lib/validate.js [hunk 7]
```diff
@@ -41,7 +41,7 @@ module.exports = {
 
     _.forEach(this.serverless.service.functions, (functionObject, functionName) => {
       (functionObject.events || []).forEach(event => {
-        if (_.has(event, 'http')) {
+        if ('http' in event) {
```
Comment (medikoo) [inline lib/plugins/aws/package/compile/events/apiGateway/lib/validate.js:44]: Let's not use `in`, this should be simply `if (event.http)`

File: lib/plugins/aws/package/compile/events/cognitoUserPool/index.js [hunk 8]
```diff
@@ -350,7 +350,10 @@ class AwsCompileCognitoUserPoolEvents {
       const userPoolLogicalId = this.provider.naming.getCognitoUserPoolLogicalId(poolName);
 
       // If overrides exist in `Resources`, merge them in
-      if (_.has(this.serverless.service.resources, userPoolLogicalId)) {
+      if (
+        this.serverless.service.resources &&
+        this.serverless.service.resources[userPoolLogicalId]
```
Comment (medikoo) [inline lib/plugins/aws/package/compile/events/cognitoUserPool/index.js:355]: I think it can be just `if (this.serverless.service.resources[userPoolLogicalId])` (?)

File: lib/plugins/aws/provider/awsProvider.js [hunk 9]
```diff
@@ -524,7 +524,7 @@ class AwsProvider {
   }
 
   getLogRetentionInDays() {
-    if (!_.has(this.serverless.service.provider, 'logRetentionInDays')) {
+    if (!('logRetentionInDays' in this.serverless.service.provider)) {
```
Comment (medikoo) [inline lib/plugins/aws/provider/awsProvider.js:527]: Let's put it as `if (!this.serverless.service.provider.logRetentionInDays)`

Comment (andreizet) [reply lib/plugins/aws/provider/awsProvider.js:527]: Hmm, this will need one more condition eventually.
In `mergeIamTemplates.test.js@633`, `logRetentionInDays` is set to 0, so `if (!0)` will be `true` and the test will fail. I should check if `logRetentionInDays` is bigger than 0, maybe add one more test case with a negative value.

File: lib/plugins/aws/package/compile/events/apiGateway/lib/validate.js (hunk 7 above)
Comment (andreizet) [reply lib/plugins/aws/package/compile/events/apiGateway/lib/validate.js:44]: Kind of similar to `lib/plugins/aws/provider/awsProvider.js`.
In `events/apiGateway/lib/validate.test.js`, `http` is set to `null` so `if (event.http)` will be false and it will not reach `this.getHttpPath(http, functionName)` which throws the error.

File: lib/plugins/aws/provider/awsProvider.js (hunk 9 above)
Comment (medikoo) [reply lib/plugins/aws/provider/awsProvider.js:527]: Let's remove problematic test. We're shifting away from such internal validation, as all will be upfront validated via config schema (handled at https://github.com/serverless/serverless/issues/6562 )

File: lib/plugins/aws/package/compile/events/apiGateway/lib/validate.js (hunk 7 above)
Comment (medikoo) [reply lib/plugins/aws/package/compile/events/apiGateway/lib/validate.js:44]: It's confusingly constructed test. Lest' replace `null` with `{}`

File: lib/plugins/aws/package/compile/events/apiGateway/lib/usagePlan.js [hunk 10]
```diff
@@ -31,10 +31,7 @@ function createUsagePlanResource(that, name) {
       that.serverless.service.service
     } ${that.provider.getStage()} stage`;
     // assign quota
-    if (
-      _.has(that.serverless.service.provider, 'usagePlan.quota') &&
-      that.serverless.service.provider.usagePlan.quota !== null
-    ) {
+    if (_.get(that.serverless.service.provider, 'usagePlan.quota', null) !== null) {
```
Comment (medikoo) [inline lib/plugins/aws/package/compile/events/apiGateway/lib/usagePlan.js:34]: Let's simplify it, to simply: `if (_.get(that.serverless.service.provider, 'usagePlan.quota')`

Note we're expecting object as `quota`, hence we should accept only truthy values

File: lib/plugins/aws/package/compile/events/apiGateway/lib/usagePlan.js [hunk 11]
```diff
@@ -46,10 +43,7 @@ function createUsagePlanResource(that, name) {
       });
     }
     // assign throttle
-    if (
-      _.has(that.serverless.service.provider, 'usagePlan.throttle') &&
-      that.serverless.service.provider.usagePlan.throttle !== null
-    ) {
+    if (_.get(that.serverless.service.provider, 'usagePlan.throttle', null) !== null) {
```
Comment (medikoo) [inline lib/plugins/aws/package/compile/events/apiGateway/lib/usagePlan.js:46]: Same here

File: lib/plugins/aws/package/compile/events/schedule/index.js [hunk 12]
```diff
@@ -69,7 +69,7 @@ class AwsCompileScheduledEvents {
               }
 
               if (Input && typeof Input === 'object') {
-                if (_.has(Input, 'body') && typeof Input.body === 'string') {
+                if (Input.body && typeof Input.body === 'string') {
```
Comment (medikoo) [inline lib/plugins/aws/package/compile/events/schedule/index.js:72]: We can simplify it just `typeof `check
